### PR TITLE
Nested Array Validity Fixes + `RowOperations`

### PR DESCRIPTION
**Pull request by Maxxen**
This PR fixes a bunch of small issues related to maintaining the validity variants for nested array types, and most notably fixes sorting of arrays containing nulled complex types such as VARCHAR or nulled nested types.

The validity of an `ARRAY`s child vector must "match" the parent vector. E.g. if an array itself is null, all the child elements corresponding to that array must also be null. Previously when de/serializing an array vector from/to the row-format during sorting we tried to maintain this invariant by doing another "validity broadcasting" phase after the parent vector is deserialized where we loop over all the child elements and invalidate them so as to match the parent validity. However, this does not work very well recursively, and in the case of e.g. arrays holding strings it breaks immediately as we end up deserializing strings that contain invalid data before we get the chance to NULL them, which causes the verification to fail.

In this PR we generalize how nested validity is handled for row-format de/serialization by abstracting the validity mask location pointer (and potential struct entry offset) behind a optional `NestedValidity` pointer, which transparently hides how the validity bits of the nested vector should be set depending on if they are within a struct or a list/array. This enables us to avoid having to invalid the children "after" deserialization in the case of arrays, which avoids a bunch of extra work, allows string verification to work properly and is also less error prone. 

The cost is introducing an extra branch when getting/setting validity during row format de/serialization but I don't think it is going to be very significant as it should be easily be predicted and I know the whole row-format is subject to change in the future anyway. We could always get rid of this branch with templating in the future as well.

I still need to fix the exact same issue in the tuple-data serialization used during joins, but will do that in a follow up PR ASAP.

### CODE REVIEW

File: src/common/row_operations/row_heap_scatter.cpp [hunk 1]
```diff
@@ -7,6 +7,81 @@ namespace duckdb {
...
+	} else {
+		// Is Struct
+		ValidityBytes row_mask(struct_validity_locations[idx]);
+		const auto valid = row_mask.RowIsValid(row_mask.GetValidityEntry(entry_idx), idx_in_entry);
+		return valid;
+	}
+}
+
+/*
+VerticalParentValidity::VerticalParentValidity(data_ptr_t validitymask_location)
```
Comment (Mytherin) [inline src/common/row_operations/row_heap_scatter.cpp:49]: Is this left here on purpose?

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! LGTM - maybe @lnkuiper can also have a look?

File: src/common/row_operations/row_heap_scatter.cpp (hunk 1 above)
Comment (Maxxen) [reply src/common/row_operations/row_heap_scatter.cpp:49]: No...

File: src/common/row_operations/row_heap_scatter.cpp [hunk 2]
```diff
@@ -7,6 +7,81 @@ namespace duckdb {
...
+    : list_validity_location(nullptr), struct_validity_locations(validitymask_locations), entry_idx(0),
+      idx_in_entry(0) {
+	ValidityBytes::GetEntryIndex(child_vector_index, entry_idx, idx_in_entry);
+}
+
+void NestedValidity::SetInvalid(idx_t idx) {
+	if (list_validity_location) {
+		// Is List
+		auto entry_offset = idx / 8;
+		auto bit_offset = idx % 8;
```
Comment (lnkuiper) [inline src/common/row_operations/row_heap_scatter.cpp:24]: Can we use this for computing the entry/bit offset?
```c++
idx_t entry_idx;
idx_t idx_in_entry;
ValidityBytes::GetEntryIndex(col_idx, entry_idx, idx_in_entry);
```

**Review by lnkuiper — CHANGES_REQUESTED** [review]
Just a small nitpick, looks good otherwise:

File: src/common/row_operations/row_heap_scatter.cpp (hunk 2 above)
Comment (Maxxen) [reply src/common/row_operations/row_heap_scatter.cpp]: Fixed!

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**